src/pspm_interpolate.m

```diff
@@ -65,238 +51,169 @@
+% 2.1 load data
+if method == 1
+    data{1}.data = indata;
+    method = 1;
+elseif method == 2
+    [lsts, infos, alldata] = pspm_load_data(indata);
+    if lsts < 1, return; end
+    [lsts, data, pos_of_channel] = pspm_select_channels(alldata, 'wave');
+    if lsts < 1, return; end
+    method = 2;
+elseif method == 3
+    [lsts, data, infos, pos_of_channel] = pspm_load_channel(indata, channel, 'wave');
+    if lsts < 1, return; end
+    data = {data};
+end
+
+% 2.2 work on all channels
+for i_channel = 1:numel(data)
+    v = data{i_channel}.data;
+    if numel(find(~isnan(v))) < 2
       warning('ID:invalid_input',...
-        'Need at least two sample points to run interpolation (Channel %i). Skipping.', k);
+        'Need at least two sample points to run interpolation (Channel %i). Skipping.', i_channel);
     else
-      x = 1:length(dat);
-      v = dat;
-      % add some other checks here if you want to filter out other data
-      % features (e. g. out-of-range values)
-      filt = isnan(v);
-      xq = find(filt);
-      % remember how many data is being interpolated
-      interp_frac(k) = numel(xq)/numel(v);
-      % throw away data matching 'filt'
-      x(xq) = [];
-      v(xq) = [];
-      % check for overlaps
-      if numel(xq) < 1
-        e_overlap = 0;
-        s_overlap = 0;
-      else
-        e_overlap = max(xq) > max(x);
-        s_overlap = min(xq) < min(x);
-      end
-      if s_overlap || e_overlap
-        if ~options.extrapolate
-          warning('ID:option_disabled', ...
-            'Extrapolating was forced because interpolate without extrapolating cannot be done');
-          vq = interp1(x, v, xq, options.method, 'extrap');
-        elseif s_overlap && strcmpi(options.method, 'previous')
-          warning('ID:out_of_range', ['Cannot extrapolate with ', ...
-            'method ''previous'' and overlap at the beginning.']);
-          return;
-        elseif e_overlap && strcmpi(options.method, 'next')
-          warning('ID:out_of_range', ['Cannot extrapolate with ', ...
-            'method ''next'' and overlap at the end.']);
-          return;
-        else
-          % extrapolate because of overlaps
-          vq = interp1(x, v, xq, options.method, 'extrap');
-        end
-      else
-        % no overlap
-        vq = interp1(x, v, xq, options.method);
-      end
-      dat(xq) = vq;
-      if inline_flag
-        channel{k} = dat;
-      else
-        channel{k}.data = dat;
+      x = 1:length(v);
+      xq = find(isnan(v));
+      if numel(xq) > 0
+          % throw away data matching 'xq'
+          x(xq) = [];
+          v(xq) = [];
+          % check for overlaps
+          if numel(xq) < 1
+            e_overlap = 0;
+            s_overlap = 0;
+          else
+            e_overlap = max(xq) > max(x);
+            s_overlap = min(xq) < min(x);
+          end
+          if s_overlap || e_overlap
+            if ~options.extrapolate
+              warning('ID:option_disabled', ...
```

Comment:
I am not sure I understand `option_disabled`, does this mean `options.extrapolate` must be true? It seems to be `option_invalid` which can be in line with `input_invalid`.